### PR TITLE
✨PROS 4: Update to C++20

### PR DESCRIPTION
**Pull request by WillXuCodes**
#### Summary:
Retarget building to C++20 for PROS 4
#375 
#### Test Plan:
Compile with  std::format or other C++20 features and make sure they work, implement whatever we need afterwards to facilitate this if need be.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,6 +16,9 @@ EXTRA_INCDIR=$(FWDIR)/libv5rts/sdk/vexv5/include
 # Directories to be excluded from all builds
 EXCLUDE_SRCDIRS+=$(SRCDIR)/tests
 
+C_STANDARD?=gnu11
```
Comment (HotelCalifornia) [inline Makefile:19]: these should be `C_STANDARD=` etc

File: common.mk [hunk 2]
```diff
@@ -24,8 +24,8 @@ wlprefix=-Wl,$(subst $(SPACE),$(COMMA),$1)
 LNK_FLAGS=--gc-sections --start-group $(strip $(LIBRARIES)) -lgcc -lstdc++ --end-group -T$(FWDIR)/v5-common.ld
 
 ASMFLAGS=$(MFLAGS) $(WARNFLAGS)
-CFLAGS=$(MFLAGS) $(CPPFLAGS) $(WARNFLAGS) $(GCCFLAGS) --std=gnu11
-CXXFLAGS=$(MFLAGS) $(CPPFLAGS) $(WARNFLAGS) $(GCCFLAGS) --std=gnu++17
+CFLAGS=$(MFLAGS) $(CPPFLAGS) $(WARNFLAGS) $(GCCFLAGS) --std=$(C_STANDARD)
```
Comment (HotelCalifornia) [inline common.mk:27]: continued from previous comment, common.mk should then set `C_STANDARD?=gnu11` etc

**Comment by noam987** [comment]
I think the azure pipeline needs to use a newer version of GCC. It is currently using the Q3 2019 release, while C++ 20 didn't release until December 2020. I would recommend migrating it to a 2021 or 2022 version

**Review by omegaStag — APPROVED** [review]
tested some C++20 features such as ``__VA_OPT__`` or ``char8_t`` on hardware, though clangd linter was kicking up a bit of a fuss it still compiled and ran on the brain

i guess if anyone has issues trying to use C++20 stuff tell them to check the C++ feature table for GCC first

**Pull request merged**